python/dfu.py

```diff
@@ -85,14 +92,21 @@ def program(self, address, dat, block_size=None):
 
   def program_bootstub(self, code_bootstub):
     self.clear_status()
-    self.erase(0x8004000)
     self.erase(0x8000000)
-    self.program(0x8000000, code_bootstub, 0x800)
+    if self._hw_gen == 3:
+      self.erase(0x8020000)
+      self.program(0x8000000, code_bootstub, 0x400)
+    else:
+      self.erase(0x8004000)
+      self.program(0x8000000, code_bootstub, 0x800)
     self.reset()
 
   def recover(self):
     from panda import BASEDIR
-    fn = os.path.join(BASEDIR, "board", "obj", "bootstub.panda.bin")
+    if self._hw_gen == 3:
+      fn = os.path.join(BASEDIR, "board", "obj", "bootstub.panda_gen3.bin")
+    else:
+      fn = os.path.join(BASEDIR, "board", "obj", "bootstub.panda.bin")
```

Comment:
might make sense to move this to a config file somewhere then we can import this here, in the SConscript, and tests